src/wasm/wasm-binary.cpp

```diff
@@ -2447,13 +2504,23 @@ void WasmBinaryBuilder::readTableElements() {
     throwError("Too many segments");
   }
   for (size_t i = 0; i < numSegments; i++) {
-    auto tableIndex = getU32LEB();
-    if (tableIndex != 0) {
+    auto headerFormat = getU32LEB();
+    if (headerFormat != 0) {
       throwError("Table elements must refer to table 0 in MVP");
     }
```

Comment:
This seems like an important restriction to lift in this PR.

Reply:
This is partially lifted now as I added support headerFormat 0x2.